virttest/qemu_vm.py

```diff
@@ -1651,7 +1652,7 @@ def sort_key(dev):
         # Some versions of windows don't support more than 2 sockets of cpu,
         # here is a workaround to make all windows use only 2 sockets.
         if (vcpu_sockets and vcpu_sockets > 2 and
-                params.get("os_type") == 'windows'):
+                params.get("os_type") == 'windows' and vcpu_sockets_force is False):
```

Comment:
If we can collect the list of unsupported Windows versions, maybe we can remove this workaround?

@leidwang @vivianQizhu, what's your opinion?

Reply:
@PaulYuuu @leidwang There are too many different Windows versions and it will change yearly, I don't think it is necessary to maintain a list for it, but instead, I would suggest to have a configurable `max_vcpu_sockets`, like: `params.get('max_vcpu_sockets', 2)`, and provide users the opportunity to update it when needed, and we might update some of the Windows guest cfgs accordingly for those ones we believe necessary.

Reply:
@vivianQizhu, this limitation is only for Windows guests, so I am not sure the name is suitable. BTW, for future windows releases, maybe the limitation is gone? For current releases, only `WinXP ~ Win11` and `Win2003 ~ Win2022`, So a list also doesn't change frequently?

Reply:
@PaulYuuu
It might varies from different editions, not only versions, so it is not that easy to main such a list, we don't really track all edition changes frequently. Besides WinXP - Win8 have already EOL thus we will not have them, and Win8.1/2012/R2 are all very closed to EOL, we will need to add them and then remove them after they are EOL. Anyway, we will not be able to maintain such a list, if it does need, we might need a owner for it who need to check them on time, both @leidwang and me will not do that, that's out of our test requirement.
You are right it is only for Windows and the default should not be `2`, how about this:
I will remove `params.get("os_type") == 'windows'` and `vcpu_sockets > 2`
```
max_vcpu_sockets = params.get('max_vcpu_sockets')
if vcpu_sockets and max_vcpu_sockets and vcpu_sockets > max_vcpu_sockets:
    vcpu_sockets =  max_vcpu_sockets
```
And we set `max_vcpu_sockets = 2` in Windows.cfg, for any other specific requirement we can set them separately. This way it seems more reasonable and also stick to guest OS specification, and does not require extra effort to maintain(which I don't see the necessity).

Reply:
Hi @vivianQizhu I think we can not make the code like this:
```
max_vcpu_sockets = params.get('max_vcpu_sockets')
if vcpu_sockets and max_vcpu_sockets and vcpu_sockets > max_vcpu_sockets:
    vcpu_sockets =  max_vcpu_sockets
```
Because previous cases not set `max_vcpu_sockets`, only set `vcpu_sockets`, and the value will be limited by this workaround. If we changed the code as you provided,the workaround will be removed,so I think we'd better keep the workaround and add a configurable parameter.

Reply:
@leidwang Not sure if I miss something, by setting `max_vcpu_sockets = 2` in Windows.cfg you would get exactly same thing with previous code.

Reply:
Okay,if so,we need to add the seting to windows.cfg too.I agree with you.